python_modules/dagster/dagster/_core/executor/base.py

```diff
@@ -36,3 +40,37 @@ def retries(self) -> RetryMode:
 
         Returns: RetryMode
         """
+
+    def get_failure_or_retry_event_after_crash(
+        self,
+        step_context: "IStepContext",
+        err_info: SerializableErrorInfo,
+        known_state: "KnownExecutionState",
+    ):
+        from dagster._core.events import DagsterEvent
+
+        # determine the retry policy for the step if needed
+        retry_policy = step_context.op_retry_policy
+        retry_state = known_state.get_retry_state()
+        previous_attempt_count = retry_state.get_attempt_count(step_context.step.key)
+        should_retry = (
+            retry_policy
+            and not step_context.retry_mode.disabled
+            and previous_attempt_count < retry_policy.max_retries
+        )
+
+        if should_retry:
+            return DagsterEvent.step_retry_event(
+                step_context,
+                StepRetryData(
+                    error=err_info,
+                    seconds_to_wait=check.not_none(retry_policy).calculate_delay(
+                        previous_attempt_count + 1
+                    ),
+                ),
```

Comment:
don't need out the gate but wonder if we will want to distinguish these retries from in-process evaluated ones. Maybe content of err_info is sufficient